SSA/Experimental/Bits/Fast/BitStream.lean

```diff
@@ -363,10 +370,86 @@ Crucially, our decision procedure works by considering which equalities hold for
 --     (∀ w, (x w + y w) = z w) ↔ (∀ w, (ofBitVec (x w)) + (ofBitVec (y w)) ) := by
 --   have ⟨h₁, h₂⟩ : True ∧ True := sorry
 --   sorry
-@[simp] theorem ofBitVec_sub : ofBitVec (x - y) = (ofBitVec x) - (ofBitVec y) := sorry
-@[simp] theorem ofBitVec_add : ofBitVec (x + y) = (ofBitVec x) + (ofBitVec y) := sorry
-@[simp] theorem ofBitVec_neg : ofBitVec (-x) = -(ofBitVec x) := sorry
-@[simp] theorem ofBitVec_not : ofBitVec (~~~ x) = ~~~ (ofBitVec x) := sorry
+
+variable {w : Nat} { x y : BitVec w} {a b a' b' : BitStream}
+
+local notation a " ≈  " b  => EqualUpTo w a b
```

Comment:
I suggest that the notation carries a $w$ inside it, so that it's clear from the syntax what the $w$ is.

Reply:
Should this not be clear from the expressions on both sides?

Reply:
Sadly not, because the $a, b$ are bitstreams which have no width.